### PR TITLE
Use QubitId in stepresult DM.

### LINKED ISSUE #1232: StepResult's denisty matrix should not take in integer indices but qubits instead.
Figuring these out from the natural interfaces of the simulation requires using the qubit ordering or looking into the qubit_map.  The more natural way to do this is to use the qubit_map.

### PR DESCRIPTION
**Pull request by MichaelBroughton**
fixes #1232 .

### CODE REVIEW

File: cirq/sim/simulator.py [hunk 1]
```diff
@@ -618,9 +618,11 @@ def density_matrix(self, indices: Iterable[int] = None) -> np.ndarray:
                 corresponding to the state.
         """
         return wave_function.density_matrix_from_state_vector(
-            self.state(), indices)
+            self.state(),
+            [self.qubit_map[q] for q in qubits] if qubits is not None else None
+        )
 
-    def bloch_vector(self, index: int) -> np.ndarray:
+    def bloch_vector(self, index: ops.QubitId) -> np.ndarray:
```
Comment (Strilanc) [inline cirq/sim/simulator.py]: Rename `index` to `qubit`.

Comment (MichaelBroughton) [reply cirq/sim/simulator.py]: changed

File: cirq/sim/simulator.py [hunk 2]
```diff
@@ -605,7 +605,7 @@ def density_matrix(self, indices: Iterable[int] = None) -> np.ndarray:
                             \end{bmatrix}
```
Comment (kevinsung) [inline cirq/sim/simulator.py]: You need to update the docstring to replace instances of `indices` with `qubits`.

**Comment by kevinsung** [comment]
@Strilanc you mean that allowing integer indices could be confusing, right? I agree, and I think it's better to only allow specifying qubits.

File: cirq/sim/simulator.py (hunk 2 above)
Comment (MichaelBroughton) [reply cirq/sim/simulator.py]: fixed it

**Pull request merged**